crates/common_utils/src/pii.rs

```diff
@@ -25,6 +26,103 @@ pub const REDACTED: &str = "Redacted";
 /// Type alias for serde_json value which has Secret Information
 pub type SecretSerdeValue = Secret<serde_json::Value>;
 
+/// Phone Number
+#[derive(Debug, serde::Deserialize, serde::Serialize)]
```

Comment:
use serde [try_from](https://serde.rs/container-attrs.html#try_from) attribute for deserialization here....
Currently Deserialize won't validate for phone numbers,

You'll need to 
```
impl TryFrom<String> for PhoneNumber
or
impl TryFrom<Secret<String>> for PhoneNumber
```

we can instead use `try_from = String/Secret<String>` that way serde deserializes it to the corresponding struct and then runs the try_from method to get the correct PhoneNumber or throw a ParsingError